service.go

```diff
@@ -161,7 +161,7 @@ func (s *Service) name() string {
 
 	w.WriteString(s.Name)
 
-	if s.Revision != "" {
+	if len([]rune(s.Revision)) > 5  {
```

Comment:
Wait! I really stuggle to get this all straight, so I'm not sure about the use of `[]rune` for this check. Can you all please help me think about this, it turns my mind to mush when I try to think about it? 

The slice `s.Revision[0:6]` _I think_ will slice based on _bytes_, in which case we want the check to be the length of the byte slice/string representation `len(s.Revision)`, not the `[]rune`. As written now, I think this will still always avoid the out-of-bounds panic because `len([]rune(x)) >= len(x)`, assuming all runes are at least one byte. But there could be cases where there are less than 6 runes, but more than 5 bytes???? In that case this setup would fail the test even though the string was safely sliceable.

All that being said, if we know we are only dealing with ascii, it might not matter anyway.

Reply:
@zdjones I think it will only be dealing with ascii!

Reply:
So then `len(s.Revision)` as it is safe as long at git stays with ascii? happy with that @zeeraw @ladydascalie

Reply:
Yes, I'm happy with that.